docs/docs/reference-glossary.md

```diff
@@ -0,0 +1,100 @@
+---
+
+id: glossary
+title: A Glossary of Terms in React
+permalink: docs/glossary.html
+
+---
+
+#A Glossary of Terms in React
+
+
+## Single Page Application
+A single page application is an application that loads a single HTML page and all the necessary assets (CSS, Images & JavaScript) required for the application to run. Any interactions with the page or subsequent pages do not require a round trip to the server which means the page is not reloaded.
+
+
+## ES6/ES2015
+ ES6 AKA ES2015 is the current version of the ECMAScript Language Specification standard, which the JavaScript language is an implementation of. It includes many additions to the previous versions such as: arrow functions, classes, template literals, and `let` and `const` statements. 
+
+## Compilers/Transpilers
+A JavaScript compiler (or transpiler) takes JavaScript code, transforms it and returns JavaScript code in a different format. The most common use case is to take ES2015/ES6 syntax and transform it into syntax that browsers are capable of interpreting. Babel is the compiler used with React. 
+
+## Bundlers
+Bundlers put all of your JavaScript code & dependency into one "bundle", usually into one file. Some bundlers commonly used in React applications include: Webpack & Browserify 
+
+## Package Manager 
+Package managers are tools that allow you to manage dependencies in your project. npm & Yarn are two package managers commonly used in React applications
+
+## CDN
+CDN stands for Content Delivery Network. CDNs deliver cached, static content from a network of servers across the globe. 
+
+## JSX
+JSX is a syntax extension to JavaScript. It is similar to a template language. When JSX gets compiled, the result is JavaScript objects called "elements". To get a basic introduction to JSX [see the docs here](https://facebook.github.io/react/docs/introducing-jsx.html) and find a more in-depth tutorial on JSX [here](https://facebook.github.io/react/docs/jsx-in-depth.html)
```

Comment:
Should the "intro to JSX" documentation be modified? In this section: [https://facebook.github.io/react/docs/introducing-jsx.html#jsx-is-an-expression-too](https://facebook.github.io/react/docs/introducing-jsx.html#jsx-is-an-expression-too) it says "After compilation, JSX expressions become regular JavaScript objects." I can change the phrasing to say something like "JSX gets compiled to `React.createElement()` calls which return plain JavaScript objects called 'React elements'." <-- Does that feel like a better description? I was trying to keep the definitions simple and refer people to the actual documentation if they wanted to dig a little deeper.